### ISSUE TITLE
Missing debug information on CI jobs print about "mark comment for a type"

### ISSUE THREAD

**Issue by alvinh77** [nicklockwood/SwiftFormat#1767]
Hey team, seems we ran into an issue that sometimes our gitlab CI jobs failed with only one statement:
`ERROR: Job failed: exit status 1` without any additional information about the violations.

What we found:
1. When we run the checks locally, it looks fine without any failed prints.
2. Seems it should be related to `mark comment for a type` as we run `mint run --no-install SwiftFormat --strict .` locally and it will add the `MARK: - .....`

Not too sure it is an issue or just expected behaviour here. 🙏

**Comment by alvinh77** [comment]
@nicklockwood thanks for the followup
I had a check on our script. it is `mint run --no-install SwiftFormat --lint --strict .`

**Comment by nicklockwood** [comment]
Status 1 is lint failure, which makes sense if the code has not been formatted. Maybe the issue is that the actual violation messages are not being passed on by your CI? I don't know enough about gitlab to know the best approach to take there though.

**Comment by alvinh77** [comment]
yea, there is no actual violation messages are being passed/displayed on by our CI job console.
Thanks for the replies here anyway 😃

**Comment by nicklockwood** [comment]
Sorry if this seemed like I was stating the obvious 😅 What I meant was, I think the problem may be that the messages aren't being passed on, rather than that SwiftFormat isn't actually producing any useful messages.

The violation messages are typically written to stderr rather than stdout, if that's any help.

**Comment by alvinh77** [comment]
all good, good to know that, will have a look 🙏

**Comment by Brett-Best** [comment]
@nicklockwood let me know if this might need to be a separate issue but the below doesn't produce a violation - not sure if the `Rules.md` doesn't have the correct default values 🤔 

```shell
brettbest@Bretts-MacBook-Pro ~/D/SwiftFormat-Reproduction [0|1]> printf "struct MyStruct {}\n" | mint run nicklockwood/SwiftFormat@0.54.2 --enable all --lint --strict stdin
Running SwiftFormat...
(lint mode - no files will be changed.)
SwiftFormat completed successfully.
```

**Comment by nicklockwood** [comment]
@Brett-Best that certainly seems like a bug - thanks for the repro case!

**Comment by nicklockwood** [comment]
@Brett-Best it turns out this is actually the expected behavior - markTypes doesn't add a `MARK` comment if there is only one top-level declaration in the file.

> @nicklockwood let me know if this might need to be a separate issue but the below doesn't produce a violation - not sure if the `Rules.md` doesn't have the correct default values 🤔
> 
> ```shell
> brettbest@Bretts-MacBook-Pro ~/D/SwiftFormat-Reproduction [0|1]> printf "struct MyStruct {}\n" | mint run nicklockwood/SwiftFormat@0.54.2 --enable all --lint --strict stdin
> Running SwiftFormat...
> (lint mode - no files will be changed.)
> SwiftFormat completed successfully.
> ```

@Brett-Best this is actually the expected behavior - markTypes doesn't add a `MARK` comment if there is only one top-level declaration in the file.

**Comment by nicklockwood** [comment]
@Brett-Best fixed in 0.54.3

@alvinh77 I'm not sure if we actually established whether the problem Brett found was the cause of your issue or not? If it's not fixed in 0.54.3 please feel free to reopen the issue
